src/shaka_scheme/system/gc/GC.hpp

```diff
@@ -0,0 +1,38 @@
+//
+// Created by Brandon Wong on 3/26/18.
+//
+
+#ifndef SHAKA_SCHEME_GC_HPP
+#define SHAKA_SCHEME_GC_HPP
+
+#include "shaka_scheme/system/gc/GCList.hpp"
+#include "shaka_scheme/system/gc/GCData.hpp"
+#include "shaka_scheme/system/base/Data.hpp"
+
+namespace shaka {
+
+    namespace gc {
+
+    /**
+         * @brief Implements the factory method of the GC, which allows the
+         * construction of GCData and automatically adds those GCData to the
+         * GCList
+         */
+        class GC {
```

Comment:
Where is `GCNode`?

Reply:
Recall that GCNode is what wraps the GCData and will ultimately act as an alias to NodePtr, thus it should not be included in the GC class. GC class simply houses the factory method which will allocate GCData objects on the heap given data, and will automatically add those GCData objects to GCList.

Reply:
The purpose of this factory idiom is to allow the creation of GCData to take place only in unison with the garbage collector, which forces the placement of the GCData object in the managed space, preventing memory leaks. create_data will only be used inside of the managed create_node lambda, to create the GCData object which will be placed inside of the GCNode to be returned to the top level. Since the idiom has been to use create_node() exclusively to create NodePtr, there should be no confusion having this create_data method defined on class GC, segregated away in the gc namespace, that is used by the new create_node to actually allocate the memory that the “NodePtr” points to. The public interface will not change.

Reply:
Alright. This is fine. Good.